noxfile.py

```diff
@@ -89,9 +89,9 @@ def test(session: nox.Session) -> None:
         shutil.rmtree(sdist_dir, ignore_errors=True)
 
     # fmt: off
-    session.install("setuptools")
+    session.install("build")
     session.run(
-        "python", "setup.py", "sdist", "--formats=zip", "--dist-dir", sdist_dir,
+        "python", "-m", "build", "--sdist", "--outdir", sdist_dir,
```

Comment:
How about preventing runpy from injecting CWD into the import path?
```suggestion
        "python", "-I", "-m", "build", "--sdist", "--outdir", sdist_dir,
```
(could also do `-E`, if the constraints allow it)